### PR TITLE
Boris wip

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: src/main/kotlin/org/jitsi/rtp/RtpTwoByteHeaderExtension.kt [hunk 1]
```diff
@@ -40,7 +40,15 @@ class RtpTwoByteHeaderExtension(val buf: ByteBuffer) : RtpHeaderExtension() {
...
-        const val COOKIE: Short = 0x1000.toShort()
+        const val COOKIE: Short = 0x1000
+
+        /**
+         *  Checks if `headerExtensionType` matches the IDs used by
+         *  [RtpTwoByteHeaderExtension]. See
+         * https://tools.ietf.org/html/rfc5285#section-4.3
+         */
+        fun headerExtensionTypeMatches(headerExtensionType: Short): Boolean
+                = COOKIE.compareTo(headerExtensionType.toInt() and 0xfff0) == 0
```
Comment (bbaldino) [inline src/main/kotlin/org/jitsi/rtp/RtpTwoByteHeaderExtension.kt]: why mask with `0xfff0`?  maybe i'm missing something in the spec, but the header extension type cookie is a full 2 bytes from what i see?

File: src/main/kotlin/org/jitsi/rtp/RtpHeader.kt [hunk 2]
```diff
@@ -252,7 +252,7 @@ open class RtpHeader {
             appendln("hasExtension: $hasExtension")
             appendln("csrcCount: $csrcCount")
             appendln("marker: $marker")
-            appendln("payloadType: $payloadType")
+            appendln("packetType: $payloadType")
```
Comment (bbaldino) [inline src/main/kotlin/org/jitsi/rtp/RtpHeader.kt]: was this change intentional? this is in rtp header

File: src/test/java/org/jitsi/rtp/RtxPacketTest.kt [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
 
+        // Extension
         0xBE.toByte(), 0xDE.toByte(), 0x00.toByte(), 0x01.toByte(),
         0x51.toByte(), 0x00.toByte(), 0x0B.toByte(), 0x00.toByte(),
-        0x17.toByte(), 0x9D.toByte(), 0x90.toByte(), 0x80.toByte(),
+
+        // OSN
+        (osn shr 8).toByte(), osn.toByte(),
+
+        0x90.toByte(), 0x80.toByte(),
```
Comment (bbaldino) [inline src/test/java/org/jitsi/rtp/RtxPacketTest.kt]: for completeness, label this chunk as the payload?

**Review by bbaldino — COMMENTED** [review]
good stuff!

File: src/main/kotlin/org/jitsi/rtp/RtpTwoByteHeaderExtension.kt (hunk 1 above)
Comment (bgrozev) [reply src/main/kotlin/org/jitsi/rtp/RtpTwoByteHeaderExtension.kt]: The cookie is 2 full bytes, but 5285 defines anything starting with 0x100 (12 bits) to indicate the two-byte format. See https://tools.ietf.org/html/rfc5285#section-4.3

File: src/main/kotlin/org/jitsi/rtp/RtpHeader.kt (hunk 2 above)
Comment (bgrozev) [reply src/main/kotlin/org/jitsi/rtp/RtpHeader.kt]: Ops, I overdid it

File: src/main/kotlin/org/jitsi/rtp/RtpTwoByteHeaderExtension.kt (hunk 1 above)
Comment (bbaldino) [reply src/main/kotlin/org/jitsi/rtp/RtpTwoByteHeaderExtension.kt]: Ah, now I see.

File: src/main/kotlin/org/jitsi/rtp/RtpHeaderExtensions.kt [hunk 4]
```diff
@@ -40,8 +36,8 @@ class RtpHeaderExtensions {
         fun parse(buf: ByteBuffer): MutableMap<Int, RtpHeaderExtension> {
             val headerExtensionType = buf.getShort()
             val headerExtensionParser = when {
-                headerExtensionType.isOneByteHeaderType() -> ::RtpOneByteHeaderExtension
-                headerExtensionType.isTwoByteHeaderType() -> ::RtpTwoByteHeaderExtension
+                RtpOneByteHeaderExtension.headerExtensionTypeMatches(headerExtensionType) -> ::RtpOneByteHeaderExtension
```
Comment (bbaldino) [inline src/main/kotlin/org/jitsi/rtp/RtpHeaderExtensions.kt]: just wondering, why do you prefer this way over the extension method?

Comment (bgrozev) [reply src/main/kotlin/org/jitsi/rtp/RtpHeaderExtensions.kt]: I just wanted to keep the logic in the extension classes

Comment (bbaldino) [reply src/main/kotlin/org/jitsi/rtp/RtpHeaderExtensions.kt]: How about marking the extension functions as private?  That's probably what it should've been anyway (also, extension functions aren't applied globally: they have to be explicitly imported so we're not 'polluting' all `Short`s with those helper functions).

I think the static function is ok, it's just so much more verbose (and I do think the extension function reads more naturally)

**Review by bbaldino — APPROVED** [review]
